### ISSUE TITLE
transitions

### ISSUE THREAD

**Issue by timoxley** [yoshuawuyts/fsm-event#2]
forgive me, I don't understand the purpose of defining transitions in your lib other than for defining which states can reach each other. i.e. 

It makes more sense in my head to use transition names rather than the states.

``` js
m.init('START') // set initial state
m('pause') // PAUSED state
```

Can you elaborate on your design? I'm not sure I understand. Not your fault, probably mine.

**Comment by yoshuawuyts** [comment]
I don't think it's your fault but rather a flaw in my design. I realized this earlier when reading through your implementations, and was going to write a PR for this today.

``` js
const fsm = require('fsm-event')

// first arg signals the start state, 
// defaulting to 'START' if none is present && 'START' exists.
const m = fsm('START', {/* states */})

m('pause')
m('data')
```

What do you think of the interface?

**Comment by timoxley** [comment]
lgtm :+1:

**Comment by yoshuawuyts** [comment]
worked on this over the weekend, but haven't been able to get a full patch. Will continue on this during the evening hours next week. Progress can be tracked in the [transitions branch](https://github.com/yoshuawuyts/fsm-event/tree/transitions). Really want to fix this ^^.

**Comment by yoshuawuyts** [comment]
released as `fsm-event@2.0.0`
